### PR TITLE
UMFPACK is needed by Orbifold parametrization

### LINKED ISSUE #3326: cgal 4.12.1 make examples failed
I  am trying to compile cgal 4.12.1 from source. It built successfully but make examples always failed with these errors
```
Scanning dependencies of target shortest_paths_multiple_sources
CMakeFiles/orbifold.dir/orbifold.cpp.o: In function `Eigen::UmfPackLU<Eigen::SparseMatrix<double, 0, int> >::~UmfPackLU()':
orbifold.cpp:(.text._ZN5Eigen9UmfPackLUINS_12SparseMatrixIdLi0EiEEED2Ev[_ZN5Eigen9UmfPackLUINS_12SparseMatrixIdLi0EiEEED5Ev]+0x1b): undefined reference to `umfpack_di_free_symbolic'
orbifold.cpp:(.text._ZN5Eigen9UmfPackLUINS_12SparseMatrixIdLi0EiEEED2Ev[_ZN5Eigen9UmfPackLUINS_12SparseMatrixIdLi0EiEEED5Ev]+0x3c): undefined reference to `umfpack_di_free_numeric'
CMakeFiles/orbifold.dir/orbifold.cpp.o: In function `CGAL::Surface_mesh_parameterization::Error_code CGAL::Surface_mesh_parameterization::Orbifold_Tutte_parameterizer_3<CGAL::Seam_mesh<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Edge_index, bool>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Vertex_index, bool> >, CGAL::Default>::computeFlattening<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Halfedge_index, CGAL::Point_2<CGAL::Simple_cartesian<double> > >, boost::associative_property_map<boost::unordered::unordered_map<CGAL::Seam_mesh<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Edge_index, bool>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Vertex_index, bool> >::vertex_descriptor, int, boost::hash<CGAL::Seam_mesh<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Edge_index, bool>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Vertex_index, bool> >::vertex_descriptor>, std::equal_to<CGAL::Seam_mesh<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Edge_index, bool>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Vertex_index, bool> >::vertex_descriptor>, std::allocator<std::pair<CGAL::Seam_mesh<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Edge_index, bool>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Vertex_index, bool> >::vertex_descriptor const, int> > > > >(CGAL::Seam_mesh<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Edge_index, bool>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Vertex_index, bool> > const&, CGAL::Eigen_sparse_matrix<double> const&, CGAL::Eigen_vector<double> const&, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Halfedge_index, CGAL::Point_2<CGAL::Simple_cartesian<double> > >, boost::associative_property_map<boost::unordered::unordered_map<CGAL::Seam_mesh<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Edge_index, bool>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Vertex_index, bool> >::vertex_descriptor, int, boost::hash<CGAL::Seam_mesh<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Edge_index, bool>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Vertex_index, bool> >::vertex_descriptor>, std::equal_to<CGAL::Seam_mesh<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Edge_index, bool>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Vertex_index, bool> >::vertex_descriptor>, std::allocator<std::pair<CGAL::Seam_mesh<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Edge_index, bool>, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Simple_cartesian<double> > >::Property_map<CGAL::SM_Vertex_index, bool> >::vertex_descriptor const, int> > > >) const':
orbifold.cpp:(.text._ZNK4CGAL29Surface_mesh_parameterization30Orbifold_Tutte_parameterizer_3INS_9Seam_meshINS_12Surface_meshINS_7Point_3INS_16Simple_cartesianIdEEEEEENS8_12Property_mapINS_13SM_Edge_indexEbEENS9_INS_15SM_Vertex_indexEbEEEENS_7DefaultEE17computeFlatteningINS9_INS_17SM_Halfedge_indexENS_7Point_2IS6_EEEEN5boost24associative_property_mapINSM_9unordered13unordered_mapINSE_17vertex_descriptorEiNSM_4hashISQ_EESt8equal_toISQ_ESaISt4pairIKSQ_iEEEEEEEENS0_10Error_codeERKSE_RKNS_19Eigen_sparse_matrixIdEERKNS_12Eigen_vectorIdEET_T0_[_ZNK4CGAL29Surface_mesh_parameterization30Orbifold_Tutte_parameterizer_3INS_9Seam_meshINS_12Surface_meshINS_7Point_3INS_16Simple_cartesianIdEEEEEENS8_12Property_mapINS_13SM_Edge_indexEbEENS9_INS_15SM_Vertex_indexEbEEEENS_7DefaultEE17computeFlatteningINS9_INS_17SM_Halfedge_indexENS_7Point_2IS6_EEEEN5boost24associative_property_mapINSM_9unordered13unordered_mapINSE_17vertex_descriptorEiNSM_4hashISQ_EESt8equal_toISQ_ESaISt4pairIKSQ_iEEEEEEEENS0_10Error_codeERKSE_RKNS_19Eigen_sparse_matrixIdEERKNS_12Eigen_vectorIdEET_T0_]+0x590): undefined reference to `umfpack_di_free_symbolic'
orbifold.cpp:(.text._ZNK4CGAL29Surface_mesh_parameterization30Orbifold_Tutte_parameterizer_3INS_9Seam_meshINS_12Surface_meshINS_7Point_3INS_16Simple_cartesianIdEEEEEENS8_12Property_mapINS_13SM_Edge_indexEbEENS9_INS_15SM_Vertex_indexEbEEEENS_7DefaultEE17computeFlatteningINS9_INS_17SM_Halfedge_indexENS_7Point_2IS6_EEEEN5boost24associative_property_mapINSM_9unordered13unordered_mapINSE_17vertex_descriptorEiNSM_4hashISQ_EESt8equal_toISQ_ESaISt4pairIKSQ_iEEEEEEEENS0_10Error_codeERKSE_RKNS_19Eigen_sparse_matrixIdEERKNS_12Eigen_vectorIdEET_T0_[_ZNK4CGAL29Surface_mesh_parameterization30Orbifold_Tutte_parameterizer_3INS_9Seam_meshINS_12Surface_meshINS_7Point_3INS_16Simple_cartesianIdEEEEEENS8_12Property_mapINS_13SM_Edge_indexEbEENS9_INS_15SM_Vertex_indexEbEEEENS_7DefaultEE17computeFlatteningINS9_INS_17SM_Halfedge_indexENS_7Point_2IS6_EEEEN5boost24associative_property_mapINSM_9unordered13unordered_mapINSE_17vertex_descriptorEiNSM_4hashISQ_EESt8equal_toISQ_ESaISt4pairIKSQ_iEEEEEEEENS0_10Error_codeERKSE_RKNS_19Eigen_sparse_matrixIdEERKNS_12Eigen_vectorIdEET_T0_]+0x5b1): undefined reference to `umfpack_di_free_numeric'
orbifold.cpp:(.text._ZNK4CGAL29Surface_mesh_parameterization30Orbifold_Tutte_parameterizer_3INS_9Seam_meshINS_12Surface_meshINS_7Point_3INS_16Simple_cartesianIdEEEEEENS8_12Property_mapINS_13SM_Edge_indexEbEENS9_INS_15SM_Vertex_indexEbEEEENS_7DefaultEE17computeFlatteningINS9_INS_17SM_Halfedge_indexENS_7Point_2IS6_EEEEN5boost24associative_property_mapINSM_9unordered13unordered_mapINSE_17vertex_descriptorEiNSM_4hashISQ_EESt8equal_toISQ_ESaISt4pairIKSQ_iEEEEEEEENS0_10Error_codeERKSE_RKNS_19Eigen_sparse_matrixIdEERKNS_12Eigen_vectorIdEET_T0_[_ZNK4CGAL29Surface_mesh_parameterization30Orbifold_Tutte_parameterizer_3INS_9Seam_meshINS_12Surface_meshINS_7Point_3INS_16Simple_cartesianIdEEEEEENS8_12Property_mapINS_13SM_Edge_indexEbEENS9_INS_15SM_Vertex_indexEbEEEENS_7DefaultEE17computeFlatteningINS9_INS_17SM_Halfedge_indexENS_7Point_2IS6_EEEEN5boost24associative_property_mapINSM_9unordered13unordered_mapINSE_17vertex_descriptorEiNSM_4hashISQ_EESt8equal_toISQ_ESaISt4pairIKSQ_iEEEEEEEENS0_10Error_codeERKSE_RKNS_19Eigen_sparse_matrixIdEERKNS_12Eigen_vectorIdEET_T0_]+0x96f): undefined reference to `umfpack_di_defaults'
orbifold.cpp:(.text._ZNK4CGAL29Surface_mesh_parameterization30Orbifold_Tutte_parameterizer_3INS_9Seam_meshINS_12Surface_meshINS_7Point_3INS_16Simple_cartesianIdEEEEEENS8_12Property_mapINS_13SM_Edge_indexEbEENS9_INS_15SM_Vertex_indexEbEEEENS_7DefaultEE17computeFlatteningINS9_INS_17SM_Halfedge_indexENS_7Point_2IS6_EEEEN5boost24associative_property_mapINSM_9unordered13unordered_mapINSE_17vertex_descriptorEiNSM_4hashISQ_EESt8equal_toISQ_ESaISt4pairIKSQ_iEEEEEEEENS0_10Error_codeERKSE_RKNS_19Eigen_sparse_matrixIdEERKNS_12Eigen_vectorIdEET_T0_[_ZNK4CGAL29Surface_mesh_parameterization30Orbifold_Tutte_parameterizer_3INS_9Seam_meshINS_12Surface_meshINS_7Point_3INS_16Simple_cartesianIdEEEEEENS8_12Property_mapINS_13SM_Edge_indexEbEENS9_INS_15SM_Vertex_indexEbEEEENS_7DefaultEE17computeFlatteningINS9_INS_17SM_Halfedge_indexENS_7Point_2IS6_EEEEN5boost24associative_property_mapINSM_9unordered13unordered_mapINSE_17vertex_descriptorEiNSM_4hashISQ_EESt8equal_toISQ_ESaISt4pairIKSQ_iEEEEEEEENS0_10Error_codeERKSE_RKNS_19Eigen_sparse_matrixIdEERKNS_12Eigen_vectorIdEET_T0_]+0x9a0): undefined reference to `umfpack_di_symbolic'
orbifold.cpp:(.text._ZNK4CGAL29Surface_mesh_parameterization30Orbifold_Tutte_parameterizer_3INS_9Seam_meshINS_12Surface_meshINS_7Point_3INS_16Simple_cartesianIdEEEEEENS8_12Property_mapINS_13SM_Edge_indexEbEENS9_INS_15SM_Vertex_indexEbEEEENS_7DefaultEE17computeFlatteningINS9_INS_17SM_Halfedge_indexENS_7Point_2IS6_EEEEN5boost24associative_property_mapINSM_9unordered13unordered_mapINSE_17vertex_descriptorEiNSM_4hashISQ_EESt8equal_toISQ_ESaISt4pairIKSQ_iEEEEEEEENS0_10Error_codeERKSE_RKNS_19Eigen_sparse_matrixIdEERKNS_12Eigen_vectorIdEET_T0_[_ZNK4CGAL29Surface_mesh_parameterization30Orbifold_Tutte_parameterizer_3INS_9Seam_meshINS_12Surface_meshINS_7Point_3INS_16Simple_cartesianIdEEEEEENS8_12Property_mapINS_13SM_Edge_indexEbEENS9_INS_15SM_Vertex_indexEbEEEENS_7DefaultEE17computeFlatteningINS9_INS_17SM_Halfedge_indexENS_7Point_2IS6_EEEEN5boost24associative_property_mapINSM_9unordered13unordered_mapINSE_17vertex_descriptorEiNSM_4hashISQ_EESt8equal_toISQ_ESaISt4pairIKSQ_iEEEEEEEENS0_10Error_codeERKSE_RKNS_19Eigen_sparse_matrixIdEERKNS_12Eigen_vectorIdEET_T0_]+0xa01): undefined reference to `umfpack_di_numeric'
orbifold.cpp:(.text._ZNK4CGAL29Surface_mesh_parameterization30Orbifold_Tutte_parameterizer_3INS_9Seam_meshINS_12Surface_meshINS_7Point_3INS_16Simple_cartesianIdEEEEEENS8_12Property_mapINS_13SM_Edge_indexEbEENS9_INS_15SM_Vertex_indexEbEEEENS_7DefaultEE17computeFlatteningINS9_INS_17SM_Halfedge_indexENS_7Point_2IS6_EEEEN5boost24associative_property_mapINSM_9unordered13unordered_mapINSE_17vertex_descriptorEiNSM_4hashISQ_EESt8equal_toISQ_ESaISt4pairIKSQ_iEEEEEEEENS0_10Error_codeERKSE_RKNS_19Eigen_sparse_matrixIdEERKNS_12Eigen_vectorIdEET_T0_[_ZNK4CGAL29Surface_mesh_parameterization30Orbifold_Tutte_parameterizer_3INS_9Seam_meshINS_12Surface_meshINS_7Point_3INS_16Simple_cartesianIdEEEEEENS8_12Property_mapINS_13SM_Edge_indexEbEENS9_INS_15SM_Vertex_indexEbEEEENS_7DefaultEE17computeFlatteningINS9_INS_17SM_Halfedge_indexENS_7Point_2IS6_EEEEN5boost24associative_property_mapINSM_9unordered13unordered_mapINSE_17vertex_descriptorEiNSM_4hashISQ_EESt8equal_toISQ_ESaISt4pairIKSQ_iEEEEEEEENS0_10Error_codeERKSE_RKNS_19Eigen_sparse_matrixIdEERKNS_12Eigen_vectorIdEET_T0_]+0xb13): undefined reference to `umfpack_di_solve'
collect2: error: ld returned 1 exit status
examples/Surface_mesh_parameterization/CMakeFiles/orbifold.dir/build.make:109: recipe for target 'examples/Surface_mesh_parameterization/orbifold' failed
make[3]: *** [examples/Surface_mesh_parameterization/orbifold] Error 1
CMakeFiles/Makefile2:27665: recipe for target 'examples/Surface_mesh_parameterization/CMakeFiles/orbifold.dir/all' failed
make[2]: *** [examples/Surface_mesh_parameterization/CMakeFiles/orbifold.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
[ 88%] Building CXX object examples/Surface_mesh_shortest_path/CMakeFiles/shortest_paths_multiple_sources.dir/shortest_paths_multiple_sources.cpp.o
[ 88%] Linking CXX executable segmentation_via_sdf_values_example
[ 88%] Built target segmentation_via_sdf_values_example
[ 88%] Linking CXX executable segmentation_from_sdf_values_LCC_example
[ 88%] Built target segmentation_from_sdf_values_LCC_example
[ 88%] Linking CXX executable segmentation_with_facet_ids_example
[ 88%] Built target segmentation_with_facet_ids_example
[ 91%] Linking CXX executable shortest_paths_multiple_sources
[ 91%] Built target shortest_paths_multiple_sources
[ 91%] Linking CXX executable segmentation_from_sdf_values_SM_example
[ 91%] Built target segmentation_from_sdf_values_SM_example
[ 91%] Linking CXX executable extract_segmentation_into_mesh_example
[ 91%] Built target extract_segmentation_into_mesh_example
[ 91%] Linking CXX executable segmentation_from_sdf_values_OpenMesh_example
[ 91%] Built target segmentation_from_sdf_values_OpenMesh_example
CMakeFiles/Makefile2:2783: recipe for target 'Installation/CMakeFiles/examples.dir/rule' failed
make[1]: *** [Installation/CMakeFiles/examples.dir/rule] Error 2
Makefile:576: recipe for target 'examples' failed
make: *** [examples] Error 2
```

Platform is Ubuntu 16.04 64 bit, Openmesh 7.2  (and tried 7.1 before), gcc 5.4

I built with the cmake command 
```
cmake -DCMAKE_BUILD_TYPE=Release -DWITH_examples=ON -DWITH_demos=ON 
-DWITH_BLAS=ON -DCUDA_TOOLKIT_ROOT_DIR=/usr/local/cuda 
-DCUDA_SDK_ROOT_DIR=$HOME/opt/cuda-samples/common 
-DBUILD_SHARED_LIBS=ON -DBUILD_TESTING=ON -DWITH_CGAL_ImageIO=ON 
-DWITH_CGAL_Qt5=ON -DWITH_GMP=ON -DWITH_LAPACK=ON -DWITH_MPFI=ON 
-DWITH_MPFR=ON -DWITH_NTL=ON -DWITH_OpenGL=ON -DWITH_ZLIB=ON  
-DWITH_QGLViewer=ON -DWITH_CGAL_Core=ON
 -DQGLVIEWER_INCLUDE_DIR=$HOME/opt/libQGLViewer/include 
-DQGLVIEWER_LIBRARY_RELEASE=$HOME/opt/libQGLViewer/lib/libQGLViewer-qt5.so 
-DCIMG_INCLUDE_DIR=$HOME/cimg -DWITH_tests=ON ..
```

Not sure if this has to do with anything but ccmake (the cmake gui) complains suitesparse-dir not found . The suitesparse dir is described as a directory that contains a cmake file for suitesparse. But suitesparse doesn't have a cmake file (I compiled suitesparse 5.2 from source it is in my $HOME but it doesn't have a cmake file)

### PR DESCRIPTION
**Pull request by sloriot**
Fix #3326 

I added the subcomponent in the cache and make sure UmfPack is present before trying to compile the example.

### CODE REVIEW

**Comment by sloriot** [comment]
@lrineau up to you do decide for the milestone. I'll rebase the branch accordingly

File: Installation/cmake/modules/FindSuiteSparse.cmake [hunk 1]
```diff
@@ -246,9 +246,9 @@ macro(SuiteSparse_FIND_COMPONENTS )
 		
 		## here we allow to find at least the include OR the lib dir and just warn if one of both missing
 		if(NOT SuiteSparse_${suitesparseCompUC}_INCLUDE_DIR AND NOT SuiteSparse_${suitesparseCompUC}_LIBRARY_RELEASE)
-			set(SuiteSparse_${suitesparseCompUC}_FOUND OFF)
+			set(SuiteSparse_${suitesparseCompUC}_FOUND OFF CACHE INTERNAL "")
 		else()
-			set(SuiteSparse_${suitesparseCompUC}_FOUND ON)
+			set(SuiteSparse_${suitesparseCompUC}_FOUND ON CACHE INTERNAL "")
```
Comment (lrineau) [inline Installation/cmake/modules/FindSuiteSparse.cmake]: It is considered a bad practice to have `<package>_FOUND` set in the cache. Why do you need that?

**Comment by lrineau** [comment]
Let's keep `master`, and let try to have it in 4.13.

File: Installation/cmake/modules/FindSuiteSparse.cmake (hunk 1 above)
Comment (lrineau) [reply Installation/cmake/modules/FindSuiteSparse.cmake]: Indeed that is not necessary: the non-cache variable set in `Find<package>.cmake` are accessible from the script calling `find_package(<package>)`.

Comment (sloriot) [reply Installation/cmake/modules/FindSuiteSparse.cmake]: Updated. Thanks!

**Comment by lrineau** [comment]
Successfully tested in [CGAL-4.13-Ic-103][*], and then merged.

  [*]: https://cgal.geometryfactory.com/CGAL/testsuite/results-4.13-Ic-103.shtml

**Pull request merged**